### PR TITLE
Add an Effect method to Player

### PR DESCRIPTION
**Pull request by Prim69**

### CODE REVIEW

**Comment by Sandertv** [comment]
What in specific would you use this for?

**Comment by Prim69** [comment]
Something like /nightvision, if the player has the effect then remove it, otherwise add it

Or maybe someone wants to display the players health a certain color based on what effect they have
image
image
image


I also wanted to implement crits and saw that Pocketmine checks if the player doesn't have blindness before applying crits.

**Comment by Sandertv** [comment]
I see the use for that. I propose we change this to a `Player.Effect(effectType) (effect, ok)` method though so you can actually get the effect itself too.

**Comment by Prim69** [comment]
That's done

**Comment by RestartFU** [comment]
> That's done

I think he meant to return the effet itself, not the Type.

**Comment by Prim69** [comment]
Oops, thanks, fixed it

**Comment by Prim69** [comment]
This is ready for a review

**Comment by Sandertv** [comment]
Consider adding the Effect method to the EffectManager so that it can be re-used and doesn't require iterating over the effects.

File: server/entity/effect.go [hunk 1]
```diff
@@ -75,6 +75,18 @@ func (m *EffectManager) Remove(e effect.Type, entity Living) {
...
+// Effect returns the effect instance and true if the entity has the effect otherwise it will return an empty effect
+// instance and false.
+func (m *EffectManager) Effect(e effect.Type) (effect.Effect, bool) {
+	m.mu.Lock()
+	existing, ok := m.effects[reflect.TypeOf(e)]
+	m.mu.Unlock()
+	if ok {
+		return existing, ok
+	}
+	return effect.Effect{}, ok
```
Comment (Sandertv) [inline server/entity/effect.go:87]: Can simplify this to `return existing, ok`

Comment (Sandertv) [reply server/entity/effect.go:87]: This PR should be ready to go once you resolve this.

File: server/entity/effect.go [hunk 2]
```diff
@@ -75,6 +75,15 @@ func (m *EffectManager) Remove(e effect.Type, entity Living) {
 	}
 }
 
+// Effect returns the effect instance and true if the entity has the effect otherwise it will return an empty effect
+// instance and false.
+func (m *EffectManager) Effect(e effect.Type) (effect.Effect, bool) {
+	m.mu.Lock()
+        existing, ok := m.effects[reflect.TypeOf(e)]
+        m.mu.Unlock()
+        return existing, ok
```
Comment (T14Raptor) [inline server/entity/effect.go:84]: Fix formatting here

After this, we can merge it

**Review by Sandertv — APPROVED** [review]
Looks good, thanks for the PR!

**Pull request merged**